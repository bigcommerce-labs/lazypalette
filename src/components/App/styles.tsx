import styled from 'styled-components';

export const StyledApp = styled.div`
    background: ${({ theme: { colors } }) => colors.background};
    align-content: flex-start;
    display: flex;
    flex-direction: column;
    height: 100%;
    position: fixed;
    width: 100%;
`;

export const Viewport = styled.div`
    display: flex;
    flex-direction: row;
    flex-grow: 1;
`;
