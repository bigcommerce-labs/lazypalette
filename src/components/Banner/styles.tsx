import styled from 'styled-components';

export const StyledBanner = styled.div`
    background-color: #FFD24D;
    text-align: center;
    width: 100%;
    line-height: 2.5rem;
    font-size: 1rem;
    font-weight: 600;
    color: #303540;
`;

export const StyledIcon = styled.span`
    span {
        margin-bottom: -5px;
    }
`;

export const StyledMessage = styled.p`
    text-align: left;

    a {
        color: ${({ theme }) => theme.colors.primary};
        display: block;
        font-weight: bold;
        margin-top: 5px;
        text-decoration: none;
    }
`;

export const StyledShareableLink = styled.button`
    background: ${({ theme }) => theme.colors.warning};
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.bold}
    font-family: ${({ theme }) => theme.typography.fontFamily.headings};
    margin: .5rem 0 .5rem 1.25rem;
    outline: none;
    border: 1px solid;
    border-radius: 2px;
    display: inline-block;
    padding: .25rem .5rem;
    line-height: 1.375rem;
    cursor: pointer;
`;
