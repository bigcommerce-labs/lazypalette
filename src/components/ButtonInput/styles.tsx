import styled from 'styled-components';

export const Wrapper = styled.div`
    margin: 0;
`;

interface ButtonProps {
    border?: boolean;
    classType?: string;
}

export const Button = styled.button.attrs<ButtonProps>({})`
    height: 2rem;
    border-radius: 2px;
    box-sizing: border-box;
    border: ${({ border, theme }) => border ? `1px solid ${theme.colors.primary}` : 'none'};
    padding: ${({ border }) => border ? '0 1rem' : '0'};
    font-family: ${({ theme }) => theme.typography.fontFamily.sans };
    font-size: 1rem;
    cursor: pointer;

    ${({ border, classType, theme }) => {
        const isPrimary = classType === 'primary';

        return `
            background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
            color: ${isPrimary ? '#FFF' : theme.colors.primary};

            :hover {
                background-color: ${isPrimary ? theme.colors.primaryHover : theme.colors.secondaryHover};
                ${!border && `
                    background-color: transparent;
                `}
                border-color: ${isPrimary ? theme.colors.brandPrimary : theme.colors.primary};
                text-decoration: ${border ? 'none' : 'underline'};
            }

            :focus {
                outline: none !important;
            }

            :disabled {
                background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
                opacity: 0.5;
            }
        `;
    }}
`;

Button.defaultProps = {
    theme: {
        colors: {
            background: '#F5F7FA',
            brandPrimary: '#273A8A',
            primary: '#3F69FE',
        },
        typography: {
            fontFamily: {
                sans: '"Source Sans Pro", "Helvetica Neue", Arial, sans-serif',
            },
        },
    },
};
