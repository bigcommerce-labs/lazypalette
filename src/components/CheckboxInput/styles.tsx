import styled from 'styled-components';

export const Container = styled.div`
    position: relative;
`;

export const Input = styled.input.attrs({
    type: 'checkbox',
})`
    ${({ theme }) => `
        opacity: 0;

        :not(:checked) + label:after {
            opacity: 0;
        }

        :not(:checked) + label {
            background: #FFF;
            border: 1px solid ${theme.colors.guideText};
            box-sizing: border-box;
        }
    `}
`;

export const Label = styled.label`
    ${({ theme }) => `
        cursor: pointer;
        position: absolute;
        width: 1.25rem;
        height: 1.25rem;
        top: 25%;
        left: 0;
        background: ${theme.colors.primary};
        border-radius: 2px;

        :after {
            content: '';
            position: absolute;
            width: .5rem;
            height: .25rem;
            background: transparent;
            top: .25rem;
            left: .25rem;
            border: 3px solid ${theme.colors.stroke};
            border-top: none;
            border-right: none;
            transform: rotate(-45deg);
        }
    `}
`;

export const HiddenLabel = styled.label`
    padding-left: .5rem;
    line-height: 1.25rem;
    display: inline-block;
`;
