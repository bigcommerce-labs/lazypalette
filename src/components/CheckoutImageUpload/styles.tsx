import styled from 'styled-components';

export const UploadButtonLabel = styled.label`
    height: 2rem;
    border-radius: 2px;
    border: 1px solid ${({ theme }) => theme.colors.primary};
    padding: 0 0.75em;
    font-size: 0.75rem;
    cursor: pointer;
    display: inline-block;
    color: ${({ theme }) => theme.colors.primary};
    margin-right: 1rem;
`;

export const CheckoutImageUploadWrapper = styled.div`
    margin: 1rem 0;
`;

export const CheckoutImageInputWrapper = styled.div`
    margin: 1rem 0;
`;

export const CheckoutImagePreviewWrapper = styled.div``;

export const CheckoutImageUploadError = styled.p`
    color: ${({ theme }) => theme.colors.error};
`;

export const CheckoutImageRemoveLink = styled.a`
    color: ${({ theme }) => theme.colors.error};
`;
