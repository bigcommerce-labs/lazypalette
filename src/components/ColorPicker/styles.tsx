import styled from 'styled-components';

export const Container = styled.div`
  margin: 0;
`;

interface SelectedColorProps {
  color: any;
}

export const SelectedColor = styled.div.attrs<SelectedColorProps>({}) `
  background: ${(props: SelectedColorProps) => `rgba(
    ${props.color.r},
    ${props.color.g},
    ${props.color.b},
    ${props.color.a}
  )`};
  border-radius: 3px;
  height: 1.5rem;
  width: 1.5rem;
  cursor: pointer;
  margin: 1px;
`;
