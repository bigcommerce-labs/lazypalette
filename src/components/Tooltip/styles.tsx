import styled from 'styled-components';

interface StyledTooltipProps {
    position: { belowElement?: boolean, x: number, y: number };
}

export const StyledTooltip = styled.div.attrs<StyledTooltipProps>({
    style: ({ position }: StyledTooltipProps) => ({
        left: `${position.x}px`,
        top: `${position.y}px`,
        transform: position.belowElement ? `translateY(0.5rem)` : `translateY(-0.5rem)`,
    }),
})`
    background: ${({ theme }) => theme.colors.primaryText};
    border-radius: 4px;
    box-shadow: ${({ theme }) => theme.elevation.floating};
    color: ${({ theme }) => theme.colors.empty};
    cursor: default;
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.thin};
    line-height: 1.5rem;
    padding: 0.5rem 1rem;
    pointer-events: auto;
    position: fixed;
    visibility: visible;
    z-index: ${({ theme }) => theme.layers.high};
}
`;

StyledTooltip.defaultProps = {
    theme: {
        colors: {
            empty: '#FFFFFF',
            primaryText: '#303540',
        },
        elevation: {
            floating: '0 2px 12px rgba(48, 53, 64, 0.2)',
        },
        layers: {
            high: 100,
        },
        typography: {
            fontSize: {
                small: '16px',
            },
            fontWeight: {
                thin: 400,
            },
        },
    },
};
