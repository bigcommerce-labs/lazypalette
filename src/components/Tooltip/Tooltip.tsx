import React, { PureComponent } from 'react';

import { StyledTooltip } from './styles';

interface TooltipProps {
    hideDelay?: number;
    message: JSX.Element | string;
    showDelay?: number;
    mock?: {
        elementClientRect: {
            bottom: number,
            left: number,
            right: number,
            top: number,
            width: number,
        },
        tooltipClientRect: {
            width: number,
            height: number,
        },
        windowWidth: number;
    };
}

interface TooltipState {
    message: (JSX.Element | string) | Array<JSX.Element | string>;
    tooltipPosition: { belowElement: boolean, x: number, y: number };
    tooltipVisible: boolean;
}

class Tooltip extends PureComponent<TooltipProps, TooltipState> {
    readonly state: TooltipState = {
        message: this.props.message,
        tooltipPosition: { belowElement: false, x: 0, y: 0 },
        tooltipVisible: false,
    };

    showTimer: NodeJS.Timer | null;
    hideTimer: NodeJS.Timer | null;
    childRef: any = React.createRef();
    tooltipRef: any = React.createRef();

    componentDidUpdate() {
        if (this.state.tooltipVisible) {
            const tooltipPosition = this.getTooltipPosition();

            if (tooltipPosition.x !== this.state.tooltipPosition.x ||
                tooltipPosition.y !== this.state.tooltipPosition.y) {
                this.setState({ tooltipPosition });
            }
        }
    }

    showTooltip = () => {
        this.setState({
            message: this.props.message,
            tooltipPosition: { belowElement: false, x: -2000, y: -2000 },
            tooltipVisible: true,
        });
    };

    getTooltipPosition = () => {
        const { mock } = this.props;

        const { bottom, left, right, top, width } = mock
            ? mock.elementClientRect
            : this.childRef.current.getBoundingClientRect();

        const { height: tooltipHeight, width: tooltipWidth } = mock
            ? mock.tooltipClientRect
            : this.tooltipRef.current.getBoundingClientRect();

        const windowWidth = mock ? mock.windowWidth : window.innerWidth;

        // get the maximum and minimum values of x that would still fully display the tooltip
        const xMax = windowWidth - tooltipWidth;
        const xMin = tooltipWidth / 2;

        // attempt to position the tooltip such that its center is  aligned with the center of the element
        let x = left + (width / 2) - (tooltipWidth / 2);

        if (x < xMin) {
            // if the tooltip goes off the left edge of the screen, instead align its left edge with the element
            x = left;
        } else if (x > xMax) {
            // if the tooltip goes off the right edge, align its right edge with the child
            x = right - tooltipWidth;
        }

        // attempt to position the tooltip above the element
        let y = top - tooltipHeight;
        let belowElement = false;

        if (y < 0) {
            // if the tooltip cannot fit above the element, position it below the element instead.
            y = bottom;
            belowElement = true;
        }

        return { belowElement, x, y };
    };

    hideTooltip = () => {
        this.setState({ tooltipVisible: false });
    };

    startShowTimer = () => {
        this.showTimer = setTimeout(() => {
            this.showTooltip();
            this.clearShowTimer();
        }, this.props.showDelay === undefined ? 250 : this.props.showDelay);
    };

    clearShowTimer = () => {
        if (this.showTimer) {
            clearTimeout(this.showTimer);
            this.showTimer = null;
        }
    };

    startHideTimer = () => {
        this.hideTimer = setTimeout(() => {
            this.hideTooltip();
            this.clearHideTimer();
        }, this.props.hideDelay === undefined ? 100 : this.props.hideDelay);

    };

    clearHideTimer = () => {
        if (this.hideTimer) {
            clearTimeout(this.hideTimer);
            this.hideTimer = null;
        }
    };

    onMouseEnter = () => {
        this.state.tooltipVisible ? this.clearHideTimer() : this.startShowTimer();
    };

    onMouseLeave = () => {
        this.state.tooltipVisible ? this.startHideTimer() : this.clearShowTimer();
    };

    render() {
        const child = React.Children.only(this.props.children);
        const { message, tooltipPosition, tooltipVisible } = this.state;

        return (
            <>
                {tooltipVisible &&
                    <StyledTooltip
                        onMouseEnter={this.onMouseEnter}
                        onMouseLeave={this.onMouseLeave}
                        innerRef={this.tooltipRef}
                        position={tooltipPosition}
                    >
                        {message}
                    </StyledTooltip>
                }
                {React.cloneElement(child, {
                    innerRef: this.childRef,
                    onMouseEnter: this.onMouseEnter,
                    onMouseLeave: this.onMouseLeave,
                })}
            </>

        );
    }
}

export default Tooltip;
