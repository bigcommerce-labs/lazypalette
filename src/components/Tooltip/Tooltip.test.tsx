import { mount } from 'enzyme';
import React from 'react';
import styled from 'styled-components';

import { StyledTooltip } from './styles';
import Tooltip from './Tooltip';

const ChildElement = styled.div``;

describe('Tooltip', () => {
    let tooltip: any;
    let mock: any;

    beforeEach(() => {
        mock = {
            elementClientRect: {
                bottom: 300,
                left: 200,
                right: 400,
                top: 200,
                width: 200,
            },
            tooltipClientRect: {
                height: 100,
                width: 100,
            },
            windowWidth: 1000,
        };
    });

    describe('renders correctly', () => {
        beforeEach(() => {
            tooltip = mount(
                <Tooltip
                    message="hello"
                    mock={mock}
                >
                    <ChildElement/>
                </Tooltip>
            );
        });

        it('when the tooltip is not shown', () => {
            expect(tooltip).toMatchSnapshot();
        });

        it('when the tooltip is shown', () => {
            tooltip.setState({ tooltipVisible: true });
            expect(tooltip).toMatchSnapshot();
        });

        it('when the tooltip is shown with a component as a message', () => {
            tooltip = mount(
                <Tooltip
                    message={<a>Click me</a>}
                    mock={mock}
                >
                    <ChildElement/>
                </Tooltip>
            );

            tooltip.setState({ tooltipVisible: true });
            expect(tooltip).toMatchSnapshot();
        });

        it('when the tooltip is shown but the message is undefined', () => {
            tooltip.setState({ message: undefined, tooltipVisible: true });
            expect(tooltip).toMatchSnapshot();
        });
    });

    describe('positions itself correctly', () => {
        it('when there is room for the tooltip in the default position', () => {
            tooltip = mount(
                <Tooltip
                    message="hello"
                    mock={mock}
                >
                    <ChildElement/>
                </Tooltip>
            );

            tooltip.setState({
                tooltipVisible: true,
            });

            expect(tooltip.state().tooltipPosition).toEqual({
                belowElement: false,
                x: 250,
                y: 100,
            });
        });

        it('when the element is on the top of the screen', () => {
            mock.elementClientRect.top = 90;

            tooltip = mount(
                <Tooltip
                    message="hello"
                    mock={mock}
                >
                    <ChildElement/>
                </Tooltip>
            );

            tooltip.setState({
                tooltipVisible: true,
            });

            expect(tooltip.state().tooltipPosition).toEqual({
                belowElement: true,
                x: 250,
                y: 300,
            });
        });

        it('when the element is on the left of the screen', () => {
            mock.elementClientRect.left = 10;
            mock.elementClientRect.width = 50;

            tooltip = mount(
                <Tooltip
                    message="hello"
                    mock={mock}
                >
                    <ChildElement/>
                </Tooltip>
            );

            tooltip.setState({
                tooltipVisible: true,
            });

            expect(tooltip.state().tooltipPosition).toEqual({
                belowElement: false,
                x: 10,
                y: 100,
            });
        });

        it('when the element is on the right of the screen', () => {
            mock.elementClientRect.left = 930;
            mock.elementClientRect.right = 980;
            mock.elementClientRect.width = 50;

            tooltip = mount(
                <Tooltip
                    message="hello"
                    mock={mock}
                >
                    <ChildElement/>
                </Tooltip>
            );

            tooltip.setState({
                tooltipVisible: true,
            });

            expect(tooltip.state().tooltipPosition).toEqual({
                belowElement: false,
                x: 880,
                y: 100,
            });
        });
    });

    describe('when handling mouse events', () => {
        beforeEach(() => {
            jest.useFakeTimers();

            tooltip = mount(
                <Tooltip
                    hideDelay={500}
                    message="hello"
                    mock={mock}
                    showDelay={1000}
                >
                    <ChildElement/>
                </Tooltip>
            );
        });

        describe('when the tooltip is not shown', () => {
            describe('when the mouse enters the child element', () => {
                it('starts a timer to show the tooltip', () => {
                    tooltip.find(ChildElement).simulate('mouseEnter');
                    expect(setTimeout).toHaveBeenCalledTimes(1);
                    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 1000);
                    jest.runAllTimers();
                    expect(tooltip.state().tooltipVisible).toEqual(true);
                });
            });

            describe('when the mouse leaves the child element', () => {
                it('stops the timer to show the tooltip', () => {
                    tooltip.find(ChildElement).simulate('mouseEnter');
                    expect(setTimeout).toHaveBeenCalledTimes(1);
                    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 1000);
                    tooltip.find(ChildElement).simulate('mouseLeave');
                    jest.runAllTimers();
                    expect(tooltip.state().tooltipVisible).toEqual(false);
                });
            });
        });

        describe('when the tooltip is shown', () => {
            beforeEach(() => {
                tooltip.setState({ tooltipVisible: true });
            });

            describe('when the mouse leaves the child element', () => {
                it('starts the timer to hide the tooltip', () => {
                    tooltip.find(ChildElement).simulate('mouseLeave');
                    expect(setTimeout).toHaveBeenCalledTimes(1);
                    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 500);
                    jest.runAllTimers();
                    expect(tooltip.state().tooltipVisible).toEqual(false);
                });
            });

            describe('when the mouse leaves the tooltip element', () => {
                it('starts the timer to hide the tooltip', () => {
                    tooltip.find(StyledTooltip).simulate('mouseLeave');
                    expect(setTimeout).toHaveBeenCalledTimes(1);
                    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 500);
                    jest.runAllTimers();
                    expect(tooltip.state().tooltipVisible).toEqual(false);
                });
            });

            describe('when the mouse enters the child element', () => {
                it('stops the timer to hide the tooltip', () => {
                    tooltip.find(ChildElement).simulate('mouseLeave');
                    expect(setTimeout).toHaveBeenCalledTimes(1);
                    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 500);
                    tooltip.find(ChildElement).simulate('mouseEnter');
                    jest.runAllTimers();
                    expect(tooltip.state().tooltipVisible).toEqual(true);
                });
            });

            describe('when the mouse enters the tooltip element', () => {
                it('stops the timer to hide the tooltip', () => {
                    tooltip.find(ChildElement).simulate('mouseLeave');
                    expect(setTimeout).toHaveBeenCalledTimes(1);
                    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 500);
                    tooltip.find(StyledTooltip).simulate('mouseEnter');
                    jest.runAllTimers();
                    expect(tooltip.state().tooltipVisible).toEqual(true);
                });
            });
        });
    });
});
