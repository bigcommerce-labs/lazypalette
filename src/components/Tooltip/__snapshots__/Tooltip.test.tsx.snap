// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly when the tooltip is not shown 1`] = `
<Tooltip
  message="hello"
  mock={
    Object {
      "elementClientRect": Object {
        "bottom": 300,
        "left": 200,
        "right": 400,
        "top": 200,
        "width": 200,
      },
      "tooltipClientRect": Object {
        "height": 100,
        "width": 100,
      },
      "windowWidth": 1000,
    }
  }
>
  <styled.div
    innerRef={
      Object {
        "current": <div
          class="sc-bwzfXH uTxCW"
        />,
      }
    }
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="sc-bwzfXH uTxCW"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    />
  </styled.div>
</Tooltip>
`;

exports[`Tooltip renders correctly when the tooltip is shown 1`] = `
<Tooltip
  message="hello"
  mock={
    Object {
      "elementClientRect": Object {
        "bottom": 300,
        "left": 200,
        "right": 400,
        "top": 200,
        "width": 200,
      },
      "tooltipClientRect": Object {
        "height": 100,
        "width": 100,
      },
      "windowWidth": 1000,
    }
  }
>
  <styled.div
    innerRef={
      Object {
        "current": <div
          class="sc-bdVaJa hrVFsJ"
          style="left: 250px; top: 100px; transform: translateY(-0.5rem);"
        >
          hello
        </div>,
      }
    }
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    position={
      Object {
        "belowElement": false,
        "x": 0,
        "y": 0,
      }
    }
    theme={
      Object {
        "colors": Object {
          "empty": "#FFFFFF",
          "primaryText": "#303540",
        },
        "elevation": Object {
          "floating": "0 2px 12px rgba(48, 53, 64, 0.2)",
        },
        "layers": Object {
          "high": 100,
        },
        "typography": Object {
          "fontSize": Object {
            "small": "16px",
          },
          "fontWeight": Object {
            "thin": 400,
          },
        },
      }
    }
  >
    <div
      className="sc-bdVaJa hrVFsJ"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "left": "0px",
          "top": "0px",
          "transform": "translateY(-0.5rem)",
        }
      }
    >
      hello
    </div>
  </styled.div>
  <styled.div
    innerRef={
      Object {
        "current": <div
          class="sc-bwzfXH uTxCW"
        />,
      }
    }
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="sc-bwzfXH uTxCW"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    />
  </styled.div>
</Tooltip>
`;

exports[`Tooltip renders correctly when the tooltip is shown but the message is undefined 1`] = `
<Tooltip
  message="hello"
  mock={
    Object {
      "elementClientRect": Object {
        "bottom": 300,
        "left": 200,
        "right": 400,
        "top": 200,
        "width": 200,
      },
      "tooltipClientRect": Object {
        "height": 100,
        "width": 100,
      },
      "windowWidth": 1000,
    }
  }
>
  <styled.div
    innerRef={
      Object {
        "current": <div
          class="sc-bwzfXH uTxCW"
        />,
      }
    }
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="sc-bwzfXH uTxCW"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    />
  </styled.div>
</Tooltip>
`;

exports[`Tooltip renders correctly when the tooltip is shown with a component as a message 1`] = `
<Tooltip
  message={
    <a>
      Click me
    </a>
  }
  mock={
    Object {
      "elementClientRect": Object {
        "bottom": 300,
        "left": 200,
        "right": 400,
        "top": 200,
        "width": 200,
      },
      "tooltipClientRect": Object {
        "height": 100,
        "width": 100,
      },
      "windowWidth": 1000,
    }
  }
>
  <styled.div
    innerRef={
      Object {
        "current": <div
          class="sc-bdVaJa hrVFsJ"
          style="left: 250px; top: 100px; transform: translateY(-0.5rem);"
        >
          <a>
            Click me
          </a>
        </div>,
      }
    }
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    position={
      Object {
        "belowElement": false,
        "x": 0,
        "y": 0,
      }
    }
    theme={
      Object {
        "colors": Object {
          "empty": "#FFFFFF",
          "primaryText": "#303540",
        },
        "elevation": Object {
          "floating": "0 2px 12px rgba(48, 53, 64, 0.2)",
        },
        "layers": Object {
          "high": 100,
        },
        "typography": Object {
          "fontSize": Object {
            "small": "16px",
          },
          "fontWeight": Object {
            "thin": 400,
          },
        },
      }
    }
  >
    <div
      className="sc-bdVaJa hrVFsJ"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "left": "0px",
          "top": "0px",
          "transform": "translateY(-0.5rem)",
        }
      }
    >
      <a>
        Click me
      </a>
    </div>
  </styled.div>
  <styled.div
    innerRef={
      Object {
        "current": <div
          class="sc-bwzfXH uTxCW"
        />,
      }
    }
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="sc-bwzfXH uTxCW"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    />
  </styled.div>
</Tooltip>
`;
